src/analysis/convex/basic.lean

```diff
@@ -230,6 +231,34 @@ open_segment_translate_preimage 𝕜 a b c ▸ image_preimage_eq _ $ add_left_su
 end add_comm_group
 end ordered_ring
 
+section linear_ordered_ring
+variables [linear_ordered_ring 𝕜]
+
+section add_comm_group
+variables [add_comm_group E] [add_comm_group F] [module 𝕜 E] [module 𝕜 F]
+
+lemma midpoint_mem_segment [invertible (2 : 𝕜)] (x y : E) :
+  midpoint 𝕜 x y ∈ [x -[𝕜] y] :=
+begin
+  rw segment_eq_image_line_map,
+  exact ⟨⅟2, ⟨inv_of_nonneg.mpr zero_le_two, inv_of_le_one one_le_two⟩, rfl⟩,
+end
+
+lemma midpoint_neg [invertible (2 : 𝕜)] (x : E) :
```

Comment:
Shouldn't that go in `linear_algebra.affine_space.midpoint`?

Reply:
Also I think this should be called `midpoint_self_neg` to make room for `midpoint_neg_self` (and both should be `simp` lemmas)

Reply:
I'll do this